### PR TITLE
Issue/371 new transforms

### PR DESCRIPTION
**Pull request by kardan**
- [x] Test plan | Unit test | Integration test
- [ ] Copyright header
- [x] Code formatting
- [ ] Documentation

### CODE REVIEW

**Comment by kardan** [comment]
Better to get this new transform deployed to test so it can be poked at for feedback.

File: backend/src/org/akvo/lumen/transformation/engine.clj [hunk 1]
```diff
@@ -198,6 +199,38 @@
...
+      (add-column tenant-conn {:table-name table-name
+                               :new-column-name new-column-name})
+      (combine-columns tenant-conn
+                       {:table-name table-name
+                        :new-column-name new-column-name
+                        :first-column (get-in op-spec ["args" "columnNames" 0])
+                        :second-column (get-in op-spec ["args" "columnNames" 1])
+                        :separator (get-in op-spec ["args" "separator"])})
+      {:success? true
+       :execution-log [(str "Combined columns")]
```
Comment (jonase) [inline backend/src/org/akvo/lumen/transformation/engine.clj]: Perhaps add the combined column names + new column name?

File: client/src/components/dataset/DataTableSidebar.jsx [hunk 2]
```diff
@@ -13,12 +14,14 @@ export default function DataTableSidebar(props) {
...
+    case 'combine-columns':
+      return <CombineColumns {...props} />;
     default:
       throw new Error(`Unknown sidebar type ${props.type}`);
   }
 }
 
 DataTableSidebar.propTypes = {
-  type: PropTypes.oneOf(['edit', 'filter', 'transformationLog']).isRequired,
+  type: PropTypes.oneOf(['edit', 'filter', 'transformationLog', 'combine-columns']).isRequired,
```
Comment (jonase) [inline client/src/components/dataset/DataTableSidebar.jsx]: We are mixing `camelCase` and `kebab-case` here. Perhaps rename to `combineColumns` (in a few places)?

File: client/src/components/dataset/sidebars/CombineColumns.jsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+              value={args.get('newColumnTitle')}
+              type="text"
+              className="titleTextInput"
+              onChange={evt => this.handleChangeNewColumnTitle(evt.target.value)}
+            />
+          </div>
+        </div>
+        <SidebarControls
+          positiveButtonText="Combine"
+          onApply={() => onApply(this.state.transformation)}
```
Comment (jonase) [inline client/src/components/dataset/sidebars/CombineColumns.jsx]: Disable `onApply` if both `columnNames` are not set?

Comment (kardan) [inline client/src/components/dataset/sidebars/CombineColumns.jsx]: Yes we should!

File: backend/src/org/akvo/lumen/transformation/engine.clj (hunk 1 above)
Comment (kardan) [inline backend/src/org/akvo/lumen/transformation/engine.clj]: Agree.

File: client/src/components/dataset/DataTableSidebar.jsx (hunk 2 above)
Comment (kardan) [inline client/src/components/dataset/DataTableSidebar.jsx]: Agree.

**Comment by jonase** [comment]
:+1: lgtm

**Pull request merged**